### ISSUE TITLE
[BUG] Mixing text and math mode with LaTeXStrings

### ISSUE THREAD

**Issue by briochemc** [JuliaPlots/Plots.jl#3203]
Most backends fail to properly render a `LaTeXString` that contains both text and math. I understand each backend treats LaTeX differently, but maybe there is a simple fix for this to work on all backends? (Maybe follow the lead of the PyPlot and PGFPlotsX backends, which work.)

## Details

MWE:

```julia
using Plots, LaTeXStrings
plot(x -> 2x + 1, title=L"The equation $y = 2x + 1$")
```

The expected behavior is for the title to show like

> The equation *y* = 2*x* + 1

That is, `"The equation"` as normal text and `y = 2x + 1` as an equation  in "math mode" with italic variables. (And  the `$` signs should not appear.)  Only the PyPlot and PGFPlotsX backends worked as expected for me (see below).

### Backends

This bug occurs on (:red_circle: = not good | :green_circle: = good)
Backend      | output |
------------------|-----|
gr  :red_circle:    | <img width=100% alt="Screen Shot 2020-12-26 at 8 16 50 pm" src="https://user-images.githubusercontent.com/4486578/103148772-4c529580-47b7-11eb-8865-75d78819d087.png">  |
pyplot        :green_circle:   |   <img width=100% alt="Screen Shot 2020-12-26 at 8 18 54 pm" src="https://user-images.githubusercontent.com/4486578/103148800-963b7b80-47b7-11eb-8dee-de1b5368a891.png">  |
plotly        :red_circle:  |  <img width=100% alt="Screen Shot 2020-12-26 at 8 23 51 pm" src="https://user-images.githubusercontent.com/4486578/103148853-46a97f80-47b8-11eb-8e66-5e600b823af6.png">  |
plotlyjs      :red_circle:     |    <img width=100% alt="Screen Shot 2020-12-26 at 8 40 48 pm" src="https://user-images.githubusercontent.com/4486578/103149110-a43ecb80-47ba-11eb-81e6-3ada219d4cbc.png"> |
pgfplotsx     :green_circle:    |   <img width=100% alt="Screen Shot 2020-12-26 at 8 43 44 pm" src="https://user-images.githubusercontent.com/4486578/103149155-0dbeda00-47bb-11eb-9ee7-70bd8e290eed.png">  |
inspectdr     :red_circle:    |   <img width=100% alt="Screen Shot 2020-12-26 at 8 42 25 pm" src="https://user-images.githubusercontent.com/4486578/103149138-df40ff00-47ba-11eb-9be6-3a444a7a9a1d.png">  |
unicodeplots :red_circle:   |  <img width=100% alt="Screen Shot 2020-12-26 at 8 26 08 pm" src="https://user-images.githubusercontent.com/4486578/103148881-98eaa080-47b8-11eb-94a0-50f9ddfbaa23.png"> | 

### Versions

- Plots.jl v1.9.1
- LaTeXStrings.jl v1.2.0
- Backend versions (`]st -m`):
    - GR v0.53.0
    - PyPlot v2.9.0
    - Plotly v0.3.0
    - PlotlyJS v0.14.0
    - PGFPlotsX v1.2.10
    - InspectDR v0.3.9
    - UnicodePlots v1.3.0

Output of `versioninfo()`:
```julia
Julia Version 1.5.3
Commit 788b2c77c1 (2020-11-09 13:37 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
```

FWIW, this came up in [this PR to UnitfulRecipes.jl](https://github.com/jw3126/UnitfulRecipes.jl/pull/35) which stemmed from [this discourse post](https://discourse.julialang.org/t/uniftulrecipes-and-latexstrings/52247).

**Comment by Ezequiel92** [comment]
This can be taken even further, and easily break GR if you use "nontraditional" glyphs. 

With this MWE 

``` 
using Plots, LaTeXStrings 
plot(x -> 2x + 1, title=L"The equation $y = 2x + \alpha$") 
``` 

you get 

![latexstrings_gr_default_font](https://user-images.githubusercontent.com/76693950/103222803-9ea0cd00-4903-11eb-93a6-67bd35e9dc99.png) 

With the accompanying warnings: 

``` 
GKS: glyph missing from current font: 73 
GKS: glyph missing from current font: 97 
GKS: glyph missing from current font: 73 
GKS: glyph missing from current font: 97 
GKS: Rectangle definition is invalid in routine SET_VIEWPORT 
GKS: glyph missing from current font: 73 
GKS: glyph missing from current font: 97 
GKS: glyph missing from current font: 73 
GKS: glyph missing from current font: 97 
GKS: glyph missing from current font: 73 
GKS: glyph missing from current font: 97 
``` 

This looks like a font problem, but it is not just that the default font misses the glyph α, as it can be seen with 

``` 
using Plots, LaTeXStrings 
plot(x -> 2x + 1, title=L"The equation $y = 2x + α$") 
``` 

which doesn't break:

![latexstrings_gr_default_font_alpha_julia](https://user-images.githubusercontent.com/76693950/103223079-3d2d2e00-4904-11eb-9390-0d2b2becc8be.png) 

The problem persists even if you use a font that one expects to have all basic latex glyphs, 

``` 
using Plots, LaTeXStrings 
plot(x -> 2x + 1, title=L"The equation $y = 2x + α$", fontfamily="Computer Modern")
``` 
produces

![latexstrings_gr_computer_modern_font](https://user-images.githubusercontent.com/76693950/103223266-9bf2a780-4904-11eb-9d89-0efc4b27743b.png) 

Maybe this is a problem decodifying the string? Maybe is misinterpreting the codepoints? 

# PGFPlotsX 

With pgfplotsx(), the first two examples give the same correct result: 

``` 
using Plots, LaTeXStrings 
pgfplotsx() 
plot(x -> 2x + 1, title=L"The equation $y = 2x + \alpha$", fontfamily="Computer Modern") 
``` 

and 

``` 
using Plots, LaTeXStrings 
pgfplotsx() 
plot(x -> 2x + 1, title=L"The equation $y = 2x + \alpha$") 
``` 

produces

![latexstrings_pgfplotsx_computer_modern_font](https://user-images.githubusercontent.com/76693950/103224478-7e264200-4906-11eb-97a9-fe16e0a04676.png) 

with no warnings. 

With the Unicode alpha symbol, it doesn’t work, but it doesn’t break either: 

``` 
using Plots, LaTeXStrings 
pgfplotsx() 
plot(x -> 2x + 1, title=L"The equation $y = 2x + α$") 
``` 

produces

![latexstrings_pgfplotsx_default_font_alpha_julia](https://user-images.githubusercontent.com/76693950/103224679-ec6b0480-4906-11eb-9aef-ab29a0122093.png) 

# A right now solution

All of the above can be bypass in a hacky way. For example, if you want GR for speed and want the LaTeX font you can do: 

``` 
using Plots, LaTeXStrings 
plot(x -> 2x + 1, title=L"\textrm{The equation} \ y = 2x + \alpha", fontfamily="Computer Modern") 
``` 

resulting in 

![test](https://user-images.githubusercontent.com/76693950/103226434-dfe8ab00-490a-11eb-98a8-3a03bf11e69f.png)

For string interpolation it gets a little more complicated, because you have to manually scape the symbols and still use LaTeXStrings, otherwise it breaks:

``` 
using Plots, LaTeXStrings 
s="\\ y = 2x + \\alpha"
plot(x -> 2x + 1, title=L"\textrm{The equation} %$s", fontfamily="Computer Modern") 
``` 

resulting in 

![test_2](https://user-images.githubusercontent.com/76693950/103226676-6f8e5980-490b-11eb-8b18-77ad857fe498.png)

# Versions 

From `julia> versioninfo()` 

``` 
Julia Version 1.5.3 
Commit 788b2c77c1 (2020-11-09 13:37 UTC) 
Platform Info: 
  OS: Windows (x86_64-w64-mingw32) 
  CPU: Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz 
  WORD_SIZE: 64 
  LIBM: libopenlibm 
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake) 
``` 

From `(@v1.5) pkg> status` 

``` 
[b964fa9f] LaTeXStrings v1.2.0 
[8314cec4] PGFPlotsX v1.2.10 
[91a5bcdd] Plots v1.9.1 
[f3b207a7] StatsPlots v0.14.17 
```

**Comment by cocoa1231** [comment]
The workaround does not work if the LaTeX is in the legend. GR backend -

```
julia> using Plots, LaTeXStrings

julia> plot(x -> 2x + 1, title=L"\textrm{The equation} \ y = 2x + \alpha", fontfamily="Computer Modern") 

julia> plot(x -> 2x + 1, title=L"\textrm{The equation} \ y = 2x + \alpha", label="\alpha", fontfamily="Computer Modern")
GKS: glyph missing from current font: 7
GKS: glyph missing from current font: 7
GKS: glyph missing from current font: 7
```

Julia information -

```
julia> versioninfo()
Julia Version 1.5.3
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)

(@v1.5) pkg> status
Status `~/.julia/environments/v1.5/Project.toml`
  [c52e3926] Atom v0.12.30
  [6e4b80f9] BenchmarkTools v0.5.0
  [336ed68f] CSV v0.8.3
  [717857b8] DSP v0.6.10
  [a93c6f00] DataFrames v0.22.5
  [31c24e10] Distributions v0.24.12
  [7a1cc6ca] FFTW v1.3.1
  [f6369f11] ForwardDiff v0.10.15
  [7073ff75] IJulia v1.23.1
  [e5e0dc1b] Juno v0.8.4
  [b964fa9f] LaTeXStrings v1.2.0
  [2b0e0bc5] LanguageServer v3.2.0
  [d96e819e] Parameters v0.12.2
  [91a5bcdd] Plots v1.10.2
  [c3e4b0f8] Pluto v0.12.20
  [d330b81b] PyPlot v2.9.0
  [b3cc710f] StaticLint v4.5.0
  [24249f21] SymPy v1.0.38
  [cf896787] SymbolServer v5.1.1
```

**Comment by Ezequiel92** [comment]
You have to use LaTeXString macro in the label too (notice the `L` in the label):

`plot(x -> 2x + 1, title=L"\textrm{The equation} \ y = 2x + \alpha", label=L"\alpha", fontfamily="Computer Modern")`

work just fine for me.

**Comment by cocoa1231** [comment]
Hello. Yes that seems to work for me too. Sorry for the mistake and thank you for the help!

**Comment by anneaux** [comment]
Is there a workaround without using the macro? Because e.g. line breaking with `\n` in the latexstring only works if you use `LaTeXString("Hello \nworld")` and not with `L"Hello \nworld"`. So how can one have a line break and text and math mode within a legend/label/title ?

**Comment by Ezequiel92** [comment]
I couldn't find a combination of commands to do that (and btw, the examples I gave in my previous comments don't longer work perfectly. In some update, `\textrm{}` and `\mathrm{}` stopped working correctly, you have to put the spaces manually now, i.e. `\mathrm{Hello \ world}`).

I think  the problem is that titles (for example) only accept either full math expressions: `"\$55\$"` or full text expressions: `"hola"`, but not mixing both. That way L"something" works (is a full math expression with \$ at the beginning and end), and `LaTeXString("Hello \nworld")` works too, as it is a full text expression. From the documentation of LaTeXString.jl: "...you can use the lowest-level constructor LaTeXString(s). The only advantage of this is that it does not automatically put $ at the beginning and end of the string... ".

So,  I don't see a way to do multiline. I don't think it is supported inside full math expressions (I tried \newline, and it doesn't work either) as you are not supposed to break line inside _inline_ math in LaTeX.

**Comment by isentropic** [comment]
This should not really be a Plots issue. It's really backend limitations. My suggestion is to use the backend of that works best for you

**Comment by isentropic** [comment]
There is a workaround for GR by enclosing normal text by $\text{...}$

**Comment by briochemc** [comment]
I saw that (raw) GR now supports this (https://twitter.com/josef_heinen/status/1455095985827352576?s=20), so I tried it again with Plots.jl (v1.23.3 with backend `GR_jll v0.62.0+0`) and it worked!

```julia
using Plots, LaTeXStrings
plot(x -> 2x + 1, title=L"The equation $y = 2x + 1$")
```

gives me

<img width="712" alt="Screen Shot 2021-11-03 at 9 42 27 am" src="https://user-images.githubusercontent.com/4486578/139961763-738fdbd3-d99d-4f9f-81fb-ee6fa39e36a0.png">

The font size between math and text mode might need some tweaking but it's nice to see this working! 

It's unclear to me who made this happen but congrats to whomever they are! 😃
